python3Packages.python3-application: 3.0.6 -> 3.0.9, add NGI team

OPNA2608 opened the pull request:
Needed for https://github.com/ngi-nix/ngipkgs/pull/1259, so adding NGI team to keep it maintained.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 committed:
python3Packages.python3-application: 3.0.6 -> 3.0.9, add NGI team

OPNA2608 committed:
python3Packages.python3-application: Add passthru.updateScript

eljamm reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 421509`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/16001618413


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>

OPNA2608 committed:
python3Packages.python3-application: Modernise

- format -> pyproject
- propagatedBuildInputs -> dependencies
- Drop meta-wide "with lib"

OPNA2608 commented on pkgs/development/python-modules/python3-application/default.nix:
```diff
@@ -28,6 +29,10 @@ buildPythonPackage rec {
+    rev-prefix = "release-";
+  };
+
   meta = with lib; {
```
Sure!

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 421509`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16089276806


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.python3-application</li>
    <li>python312Packages.python3-application.dist</li>
    <li>python313Packages.python3-application</li>
    <li>python313Packages.python3-application.dist</li>
  </ul>
</details>

GaetanLepage reviewed (approved):
Thanks!
